code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,590 @@
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
+  
+servers:
+  - url: '{apiRoot}/{basePath}'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: location-subscription/v0
+        description: Base path for the device location subscription event API
+paths:
+  /event-subscriptions:
+    post:
+      tags:
+        - Location Event Subscription
+      summary: 'Create a location event subscription for a device'
+      operationId: createEventSubscription
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateEventSubscription'
+        required: true
+      callbacks:
+        event-notifications:
+          '{$request.body#/webhook/notificationUrl}':
+            post:
+              tags:
+                - Notification Callback
+              summary: "Event Subscription notification callback"
+              operationId: postEventNotification
+              requestBody:
+                required: true
+                content:
+                  application/json:
+                    schema:
+                      $ref: "#/components/schemas/EventNotifications"
+              responses:
+                '204':
+                  description: Successful event notification
+                '400':
+                  $ref: '#/components/responses/Generic400'
+                '401':
+                  $ref: '#/components/responses/Generic401'
+                '403':
+                  $ref: '#/components/responses/Generic403'
+                '500':
+                  $ref: '#/components/responses/Generic500'
+                '503':
+                  $ref: '#/components/responses/Generic503'
+      responses:
+        '201':
+          description: Created (Successful creation of event subscription)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EventSubscriptionInfo'
+        '400':
+          $ref: '#/components/responses/Generic400'
+        '401':
+          $ref: '#/components/responses/Generic401'
+        '403':
+          $ref: '#/components/responses/Generic403'
+        '409':
+          $ref: '#/components/responses/Generic409'
+        '500':
+          $ref: '#/components/responses/Generic500'
+        '503':
+          $ref: '#/components/responses/Generic503'
+    get:
+      tags:
+        - Location Event Subscription
+      summary: 'Operation to retrieve a list of event subscriptions - could be an empty list.'
+      operationId: getEventSubscriptionList
+      description: Operation to retrieve a list of event subscriptions
+      responses:
+        '200':
+          description: The list of event subscriptions is retrieved.
+          content:
+            application/json:
+              schema:
+                type: array
```

Comment:
Better create a SubscriptionInfoList type

Reply:
I changed EventSubscriptionInfo to SubscriptionInfo, actually here SubscriptionInfoList does not make sence, only SubscriptionInfo is enough